internal/bundle/bundles/apps/template/databricks.yml.tmpl

```diff
@@ -0,0 +1,42 @@
+bundle:
+  name: basic
+
+workspace:
+  root_path: "~/.bundle/{{.unique_id}}"
+
+resources:
+  apps:
+    test_app:
+      name: "{{.app_id}}"
+      description: "App which manages job created by this bundle"
+      source_code_path: ./app
+      config:
+        command:
+          - flask
+          - --app
+          - app
+          - run
+        env:
+          - name: JOB_ID
+            valueFrom: "app-job"
```

Comment:
This bit will be tricky; I expect users to try to use `${resources.jobs.foo.id}` directly here.

It will only be possible to use regular variable interpolation here and not anything from resources.

Should we add a validation rule that confirms that nothing in the `config` payload refers to `${resources.*}`?

Reply:
And if they do, this validation can instruct them to use the `resources` block below (really needs a new name) in combination with `valueFrom`.

Reply:
I'm also wondering about the pattern where someone would grant the app's service principal the `CAN_RUN` permission on a job directly. I suspect that to be a valid alternative to `resources` + `valueFrom`.

Reply:
Changed to interpolate any variable reference in the config, so now both patterns are supported https://github.com/databricks/cli/pull/1928/commits/f7a808ed494e5ffd172b56782efbe78ebf835dc9